### DISCUSSION TITLE
Confusion with GraphQL Tools / Schema Merging

### DISCUSSION THREAD

**Discussion by nguyenyou** [Urigo/graphql-modules#1802] [Q&A] [answered]
As the documentation of GraphQL Tools said: [Originally implemented in graphql-modules.](https://www.graphql-tools.com/docs/schema-merging#merging-type-definitions)

My question is: "What is the difference between graphql-modules and graphql-tools/schema-merge?

**Comment by dotansimha** [comment, marked as answer]
`graphql-modules` is implementing schema-merging concept, with added developer tools (like dependency injection, simple lifecycle based on GraphQL, and the concept of modules separation and validation), and it depends on graphql-tools.
